### PR TITLE
[Android] Fix bug. Children pages in NavigationPage were got `appearing events` even if they were not latest in navigation stack

### PR DESCRIPTION
**Pull request by rotorgames**
### Description of Change ###

If add the `NavigationPage` in an UI stack and push two pages, later delete the `NavigationPage` from the UI stack and add it again, all pages in the `NavigationStack` will have `OnAppearing` event, even if they will not be latest in `NavigationStack`. Also if do to `PopAsync`, hidden pages will not get `OnAppearing` event.

### Bugs Fixed ###

- https://bugzilla.xamarin.com/show_bug.cgi?id=52458

### API Changes ###

None

### Behavioral Changes ###

None

### PR Checklist ###

- [x] Has tests (if omitted, state reason in description)
- [x] Rebased on top of master at time of PR
- [x] Changes adhere to coding standard
- [x] Consolidate commits as makes sense

### CODE REVIEW

**Comment by rotorgames** [comment]
@hartez @jassmith @StephaneDelcroix Any news about it? I can not use the single instance of `NavigationPage` many times because all pages in it will be get OnAppearing and OnDesappearing even if they will not be the last in `NavigationStack`.

I want to see it in v2.3.4 so much. Please review it.

**Comment by rotorgames** [comment]
@rmarinho Hello. Will we be able to see it in new release soon?

File: Xamarin.Forms.Platform.Android/AppCompat/NavigationPageRenderer.cs [hunk 1]
```diff
@@ -589,6 +585,46 @@ void SetupToolbar()
 			_toolbar = bar;
 		}
 
+		void InsertPages(IEnumerable<Page> pages)
```
Comment (rotorgames) [inline Xamarin.Forms.Platform.Android/AppCompat/NavigationPageRenderer.cs]: Ok. I will try to do it soon

**Comment by StephaneDelcroix** [comment]
no activity. closing.

reopen (or open a new one) when the comments are addressed

**Pull request closed without merging**